Formula/duckdb.rb

```diff
@@ -17,21 +19,27 @@ class Duckdb < Formula
 
   depends_on "cmake" => :build
   depends_on "python@3.9" => :build
-  depends_on "utf8proc"
```

Comment:
Why do you say this is unnecessary?

`duckdb` currently has linkage with `utf8proc`, so this seems necessary to me:
```
System libraries:
  /usr/lib/libSystem.B.dylib
  /usr/lib/libc++.1.dylib
Homebrew libraries:
  /usr/local/opt/utf8proc/lib/libutf8proc.2.dylib (utf8proc)
```

Reply:
Hm. It shouldn't be linking to that  and the build bot managed to build it without the dependency. They have their own static version of utf8proc in the repository they link to.

```bash
$ otool -L /opt/homebrew/Cellar/duckdb/0.3.1_1/bin/duckdb
/opt/homebrew/Cellar/duckdb/0.3.1_1/bin/duckdb:
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1200.3.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)
```

Reply:
> They have their own static version of utf8proc in the repository they link to.

Yes, we don't want that. See https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae. This formula should be vendoring fewer dependencies, not more.

Reply:
Unfortunately, the codebase has a couple dozen third party libraries in it so it can be linked statically. It isn't just utf8proc, but also re2, zstd, thrift, snappy, sqlite, part of libpg and more.

Perhaps the whole formula should be pulled from Homebrew then?

Reply:
I've gone through the code further. Duckdb uses a modified, privately namespaced version of utf8proc, so even if yours was linked against Homebrew's `utf8proc`, it shouldn't use it.